src/vectorDBManager.ts

```diff
@@ -79,56 +109,105 @@ class VectorDBManager {
     }
   }
 
-  public static async getMemoryVectors(docHash: string): Promise<MemoryVector[] | undefined> {
+  public static async loadFile(noteFile: NoteFile, embeddingsAPI: Embeddings) {
     if (!this.db) throw new Error("DB not initialized");
+    const textSplitter = new RecursiveCharacterTextSplitter({ chunkSize: 5000 })
```

Comment:
I picked a number that vaguely worked. I reckon this should be a variable that users can change. What do you think a good default value would be?

Reply:
Did 1000 not work when you tried? Then I guess 3000 could be a safe number to start with since most embedding models should be able to handle it. 

Letting the user set it is also one way, but could be out of scope for this PR. This is a key param for RAG tuning, so I guess if people open issues on failure cases in the future I can just tell them to tune it themselves lol!

Reply:
I quite like that as an option if we're happy keeping it simple for now. It's nice to not over-optimise early on 😅